panther_hardware_interfaces/src/roboteq_driver.cpp

```diff
@@ -100,110 +98,84 @@ bool RoboteqDriver::WaitForBoot()
-  }
-}
-
-std::uint16_t RoboteqDriver::ReadVoltage()
-{
-  try {
-    return SyncSDORead<std::uint16_t>(
-      RoboteqCANObjects::voltage.id, RoboteqCANObjects::voltage.subid);
-  } catch (const std::runtime_error & e) {
-    throw std::runtime_error("Error when trying to read voltage: " + std::string(e.what()));
-  }
-}
-
-std::int16_t RoboteqDriver::ReadBatAmps1()
-{
-  try {
-    return SyncSDORead<std::int16_t>(
-      RoboteqCANObjects::bat_amps_1.id, RoboteqCANObjects::bat_amps_1.subid);
-  } catch (const std::runtime_error & e) {
-    throw std::runtime_error("Error when trying to read bat amps 1: " + std::string(e.what()));
-  }
-}
-
-std::int16_t RoboteqDriver::ReadBatAmps2()
-{
-  try {
-    return SyncSDORead<std::int16_t>(
-      RoboteqCANObjects::bat_amps_2.id, RoboteqCANObjects::bat_amps_2.subid);
-  } catch (const std::runtime_error & e) {
-    throw std::runtime_error("Error when trying to read bat amps 2: " + std::string(e.what()));
-  }
-}
-
-RoboteqDriverFeedback RoboteqDriver::ReadRoboteqDriverFeedback()
-{
-  RoboteqDriverFeedback fb;
+  RoboteqMotorsStates states;
 
   // already does locking when accessing rpdo
-  fb.motor_1.pos =
+  states.motor_1.pos =
     rpdo_mapped[RoboteqCANObjects::position_1.id][RoboteqCANObjects::position_1.subid];
-  fb.motor_2.pos =
+  states.motor_2.pos =
     rpdo_mapped[RoboteqCANObjects::position_2.id][RoboteqCANObjects::position_2.subid];
 
-  fb.motor_1.vel =
+  states.motor_1.vel =
     rpdo_mapped[RoboteqCANObjects::velocity_1.id][RoboteqCANObjects::velocity_1.subid];
-  fb.motor_2.vel =
+  states.motor_2.vel =
     rpdo_mapped[RoboteqCANObjects::velocity_2.id][RoboteqCANObjects::velocity_2.subid];
 
-  fb.motor_1.current =
+  states.motor_1.current =
     rpdo_mapped[RoboteqCANObjects::current_1.id][RoboteqCANObjects::current_1.subid];
-  fb.motor_2.current =
+  states.motor_2.current =
     rpdo_mapped[RoboteqCANObjects::current_2.id][RoboteqCANObjects::current_2.subid];
 
-  fb.fault_flags = GetByte(
-    static_cast<int32_t>(rpdo_mapped[RoboteqCANObjects::fault_script_flags.id]
-                                    [RoboteqCANObjects::fault_script_flags.subid]),
-    0);
-  fb.script_flags = GetByte(
-    static_cast<int32_t>(rpdo_mapped[RoboteqCANObjects::fault_script_flags.id]
-                                    [RoboteqCANObjects::fault_script_flags.subid]),
-    2);
+  std::unique_lock<std::mutex> lck_p(position_timestamp_mtx_);
+  states.pos_timestamp = last_position_timestamp_;
 
-  fb.runtime_stat_flag_motor_1 = GetByte(
-    static_cast<int32_t>(
-      rpdo_mapped[RoboteqCANObjects::motor_flags.id][RoboteqCANObjects::motor_flags.subid]),
-    0);
-  fb.runtime_stat_flag_motor_2 = GetByte(
-    static_cast<int32_t>(
-      rpdo_mapped[RoboteqCANObjects::motor_flags.id][RoboteqCANObjects::motor_flags.subid]),
-    1);
+  std::unique_lock<std::mutex> lck_sc(speed_current_timestamp_mtx_);
+  states.vel_current_timestamp = last_speed_current_timestamp_;
```

Comment:
We shouldn't unlock lck_p? Or meaby use scoped_lock for multiple mutexes simultaneously?

Reply:
I would prefer to leave separate locking, I changed locks to `lock_guard` and used shorter scope.